Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,708 @@
+	local __val__ = first - second;
+	O = sborrow(first, second);
+	C = first < second; 
+	resultFlags(__val__);
+	first = __val__;
+}
+
+################################################################
+
+define pcodeop TerminateCurrentInterrupt;
+define pcodeop SoftwareInterrupt;
+
+################################################################
+
+:ADD addr16, reg0_2 is opcode3_9=0x0058 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	addition(reg0_2, *:2 addr16);
+}
+
+:ADCR reg0_2 is opcode3_9=0x0005 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local oldC = zext(C);
+	addition(reg0_2, oldC);
+}
+
+:ADDR reg3_5, reg0_2 is opcode6_9=0x0003 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	addition(reg0_2, reg3_5);
+}
+
+:AND addr16, reg0_2 is opcode3_9=0x0070 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 & (*:2 addr16);
+	resultFlags(reg0_2);
+}
+
+:ANDR reg3_5, reg0_2 is opcode6_9=0x0006 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 & reg3_5;
+	resultFlags(reg0_2);
+}
+
+:B branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=0 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto branchdestf16;
+}
+
+:B branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=0 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto branchdestb16;
+}
+
+:BC branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=1 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (C == 1) goto branchdestf16;
+}
+
+:BC branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=1 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (C == 1) goto branchdestb16;
+}
+
+:BEQ branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=4 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (Z == 1) goto branchdestf16;
+}
+
+:BEQ branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=4 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (Z == 1) goto branchdestb16;
+}
+
+:BESC branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=15 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S == C) goto branchdestf16;
+}
+
+:BESC branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=15 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S == C) goto branchdestb16;
+}
+
+:BEXT branch_condition, branchdestf16 is opcode6_9=0x0008 & branch_external=1 & branch_sign=0 & branch_condition; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto branchdestf16;
```

Comment:
Should branch_condition be used here?

Reply:
Maybe the operands should be swapped (depends on the assembler being used - GI's seems to use (destination, condition), whilst a couple others do the other way around) - but yes, the bottom 4 bits of the opcode are supposed to be there as they are the bitmask to match for 4 pins on the CPU (pins EBCA0...EBCA3 when EBCI is low).  Maybe I should give this another name rather than `branch_condition` as it already matched the range I needed.

Reply:
Was just checking that you weren't missing something like `if (branch_condition) goto branchdest`